Added LikeFeedbackBox back to ArticleReader. Changed prompt message.

oskaitu opened the pull request:
Added the already existing LikeFeedbackBox back to ArticleReader. Proposed change to prompt message to user: "Would you like more articles like this?".

Tested by going into the route: "http://localhost:3000/render/read/article?id=22"

<img width="393" alt="image" src="https://github.com/zeeguu/web/assets/100945285/a923ae1f-c6ce-447e-823f-90de29e70b93">

oskaitu commented on src/reader/LikeFeedbackBox.js:
```diff
@@ -18,9 +18,9 @@ export default function LikeFeedbackBox({
   return (
     <s.FeedbackBox>
-      <small>{strings.helpUsMsg}</small>
+      {/*<h4>{strings.helpUsMsg}</h4>*/}
```
Proposal: We think this is too much of a prompt to read for the user.

mircealungu replied:
I like the short message! 

I would beta-test it with users maybe? Because the semantics of 
- Did you enjoy the article?, and
- Would you like more articles like this?

are different. 

What I'm not sure about the second is whether:
- the user will understand what is meant by "this"
- asking me whether i liked something is an easier question - because i know if i liked it or not. asking me whether i want more is a more difficult question, e.g. i might have liked it, but not want more.

oskaitu replied:
Good points about the semantics of our proposal. 
It is true that it might be a bit ambiguous for the user to interpret what "this" is in the context of the article. As it could be the topic, the difficulty level or something else.

It is a more difficult choice to make. We thought that the user might be a bit more incentivized to click on something that would have an effect on their experience. But we agree that, it is better to keep it simple.

We will revert the changes back to the initial message 😊

mircealungu replied:
Original was too long. New was ambiguous. Maybe there could be one that's concise and unambiguous? 

Hegel: thesis, antithesis => synthesis :)

oskaitu commented:
Added the functionality to send the articles that the user has seen before clicking on an article.

<img width="463" alt="articleClicked" src="https://github.com/zeeguu/web/assets/100945285/67e4b06f-8cdb-4f72-b422-518113951f33">

It updates the backend by using the logUserActivity function.

<img width="521" alt="user_activity_data1" src="https://github.com/zeeguu/web/assets/100945285/f6cd877e-cca1-45b9-99c2-f353e9d4d1a7">

Added back the original message to LikeFeedbackBox with the original message and styling on it. 

Regarding the texts, we have some suggestions, that we can talk about to our meeting on Monday :)

<img width="376" alt="likebox" src="https://github.com/zeeguu/web/assets/100945285/a9a14977-915a-4b67-8f13-60bb945dc172">

oskaitu commented:
Combined LikeFeedbackBox and DifficultyFeedbackBox together within a "box" and added the text we talked about.

<img width="389" alt="boxescombined" src="https://github.com/zeeguu/web/assets/100945285/c18e487e-9eef-4e06-8fbb-e80c70830f77">

Added source prop to LikeFeedbackBox. UMR_SOURCE is sent from web and EXTENSION_SOURCE is sent from extension.

Made the information about the articles seen before a list instead of json format.

<img width="523" alt="listofindexs" src="https://github.com/zeeguu/web/assets/100945285/e4ecd4d7-f319-4592-a312-6c1030a85ab3">

tfnribeiro reviewed (changes requested):
We are almost there!

There is something wrong with the rendering in small screens, the old feedback is getting displayed outside the screen margins.

We also shouldn't store twice the article id that we click on, as that duplicates the data, e.g.: 

![image](https://github.com/zeeguu/web/assets/17390076/1b15b505-ca4f-441d-99f0-42d899b6fa5f)

![image](https://github.com/zeeguu/web/assets/17390076/aba09559-da13-479f-bd45-29a7145b9a36)

Finally, make sure to merge the last version of the main branch with yours, as currently there are some merge conflicts so we can't directly update the main branch. If you have difficulties, let me know and I can help you, but it would be a good exercise :)

oskaitu commented on src/reader/ArticleReader.js:
```diff
@@ -264,7 +267,25 @@ export default function ArticleReader({ api, teacherArticleID }) {
+          setArticleInfo={setArticleInfo}
+          source={UMR_SOURCE}
+        />
+        <DifficultyFeedbackBox api={api} articleID={articleID} />
```
This is now fixed.

<img width="367" alt="Screenshot 2024-03-12 160830" src="https://github.com/zeeguu/web/assets/100945285/aef839fe-e92a-4c11-943f-36775f5b8f2c">

oskaitu commented on src/reader/ArticleReader.js:
```diff
@@ -264,7 +267,25 @@ export default function ArticleReader({ api, teacherArticleID }) {
-      <DifficultyFeedbackBox api={api} articleID={articleID} />
+      <s.FeedbackBox>
+        <h4
+          style={{
```
Added the style elements inside the FeedbackBox 😎

tfnribeiro reviewed (changes requested):
Almost there! Some of the changes seem to have affected the rendering of the Exercise box:

**New version from PR**:
![image](https://github.com/zeeguu/web/assets/17390076/28cb2c90-95be-46d1-8533-0a3fb452ad25)

**zeeguu.org:**
![image](https://github.com/zeeguu/web/assets/17390076/5d131cca-4cdb-4441-a232-e974b5de1aef)

oskaitu commented:
Have merged the updates from zeeguu/web into our branch. Have run the code and made appropriate changes to the functions we have added.

Not sure if relevant. But the way I was testing before was on the url: http://localhost:3000/render/read/article?id=206213
After merging changes from zeeguu/web I experience this error:
<img width="551" alt="scrollerror" src="https://github.com/zeeguu/web/assets/100945285/dc400c14-a57c-485e-a7e1-989ceaa0c1c2">

But it works fine when I use the url: http://localhost:3000/read/article?id=206213

Is it something I should look into, or should I just keep using the working path?

oskaitu commented:
Elephant is back at his place now :) 
![image](https://github.com/zeeguu/web/assets/100945285/67e47885-4df8-40cf-adbd-4d04d0b2ec47)

tfnribeiro commented:
> Have merged the updates from zeeguu/web into our branch. Have run the code and made appropriate changes to the functions we have added.
> 
> Not sure if relevant. But the way I was testing before was on the url: http://localhost:3000/render/read/article?id=206213 After merging changes from zeeguu/web I experience this error: <img alt="scrollerror" width="551" src="https://private-user-images.githubusercontent.com/100945285/312134618-dc400c14-a57c-485e-a7e1-989ceaa0c1c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyNTcwNjIsIm5iZiI6MTcxMDI1Njc2MiwicGF0aCI6Ii8xMDA5NDUyODUvMzEyMTM0NjE4LWRjNDAwYzE0LWE1N2MtNDg1ZS1hN2UxLTk4OWNlYWEwYzFjMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxMlQxNTE5MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YmIxNDhmZmRkNzRiN2M4OGIwYWY3NTIxNDgwZmI1NTlhMzllNDZjMzEyNzk1Y2EwNjJiOTQxYzIzYmQyYjZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.No5Xx2lS-PzpbgCEHiG0kRH3KPMJ-zOccG_Zs3vqMkQ">
> 
> But it works fine when I use the url: http://localhost:3000/read/article?id=206213
> 
> Is it something I should look into, or should I just keep using the working path?

This is because the scrollTop is an element that is rendered with the sidebar, so I wouldn't worry too much that it throws errors when used with the render. I will deploy an update so it doesn't throw errors to avoid it showing the error when you use the render view.

tfnribeiro reviewed (approved):
All looks good! I believe we can start collecting user data!

mircealungu commented:
Nice! I've deployed the current version to zeeguu.org. 

While testing it, I played a bit around with the look and the text and came up with this:

<img width="501" alt="image" src="https://github.com/zeeguu/web/assets/464519/5d1f8b85-93b4-4e8c-830c-76efc62a73a7">

The difference with respect to the current solution are: 
- instructions stand out less than the questions, so the learner can focus on the questions
- the instruction is directly expressed in terms of the benefits that the user gets, not in terms of zeeguu's benefits

For the first, the style changes are just use <p> instead of <h3> and add some padding left/right:
<img width="585" alt="image" src="https://github.com/zeeguu/web/assets/464519/4de1c044-2b38-441c-8ba3-a07816e8daae">

If you like this, feel free to send a quick new PR and I'll merge it.
